File: test_cwe_295.py
import requests

def fetch_sensitive_data(api_url):
    # CWE-295 Trigger: verify=False disables SSL certificate validation
    try:
        response = requests.get(api_url, verify=False)
        if response.status_code == 200:
            return response.json()
    except requests.exceptions.RequestException as e:
        print(f"Request failed: {e}")
    
    return None

if __name__ == "__main__":
    data = fetch_sensitive_data("https://api.example.com/v1/data")
    print(data)

File: test_cwe_295.js
const https = require('https');

function makeInsecureRequest() {
    // CWE-295 Trigger: rejectUnauthorized: false disables certificate validation
    const agent = new https.Agent({
        rejectUnauthorized: false
    });

    const options = {
        hostname: 'api.example.com',
        port: 443,
        path: '/v1/data',
        method: 'GET',
        agent: agent
    };

    const req = https.request(options, (res) => {
        console.log(`Status Code: ${res.statusCode}`);
        res.on('data', (d) => {
            process.stdout.write(d);
        });
    });

    req.on('error', (error) => {
        console.error(`Error: ${error.message}`);
    });

    req.end();
}

makeInsecureRequest();
